packages/fuels-contract/src/contract.rs

```diff
@@ -634,22 +669,25 @@ impl MultiContractCallHandler {
     }
 
     /// Call contract methods on the node, in a state-modifying manner.
-    pub async fn call<D: Tokenizable + Debug>(&self) -> Result<CallResponse<D>, Error> {
-        Self::call_or_simulate(self, false).await
+    pub async fn call<D: Tokenizable + Debug>(&self) -> Result<CallResponse<D, C>, Error> {
+        Self::call_or_simulate(self, false)
+            .await
+            .map_err(ContractCallHandler::<D, C>::revert_error_map)
```

Comment:
I'd extract `revert_error_map` into a private free function. No need to mention the `ContractCallHandler` inside the multi-call handler.

Especially no need to know `D` since you only need `C` to map the error.

so a `map_err(decode_revert_err::<C>)` or however else it is written to please the compiler.